build-plugin/action.yml

```diff
@@ -99,6 +122,16 @@ runs:
       with:
         subject-path: ${{ steps.package-plugin.outputs.archive }}
 
+    - name: Commit updated changelog
+      if: ${{ env.USE_CHANGELOG_GENERATOR }}
+      uses: EndBug/add-and-commit@v9
+      with:
+        add: 'CHANGELOG.md'
+        message: 'docs: update changelog for ${{ steps.package-plugin.outputs.plugin-version }} [skip ci]'
+        default_author: github_actions
+        push: true
+        branch: main
```

Comment:
Will this require additional configuration for protected branches?

Reply:
Yes, for protected branch plugin devs have to provide additional token to be able to push generated changelog. Because this feature is opt-in, devs will need to add use_changelog_generator input and the token in case of protected branch